src/base/bittorrent/session.cpp

```diff
@@ -1444,12 +1444,16 @@ void Session::configureNetworkInterfaces(lt::settings_pack &settingsPack)
         const QHostAddress addr {ip};
         if (!addr.isNull())
         {
-            const QString ip = ((addr.protocol() == QAbstractSocket::IPv6Protocol)
-                          ? ('[' + Utils::Net::canonicalIPv6Addr(addr).toString() + ']')
+            const bool isIPv6 = addr.protocol() == QAbstractSocket::IPv6Protocol;
```

Comment:
I would still add parentheses on the right hand side: `(addr.protocol() == QAbstractSocket::IPv6Protocol);`